src/dp.rs

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! Differential privacy (DP) primitives.
+use crate::field::FieldElement;
+use crate::vdaf::prg::SeedStream;
+use std::fmt::Debug;
+
+/// Positive rational number to represent DP parameters and noise distributions.
+#[allow(dead_code)]
+#[derive(Clone, Debug)]
+pub struct Rational {
+    numerator: u32,
+    denominator: u32,
+}
+
+impl Rational {
+    #[allow(dead_code)]
+    fn new(numerator: u32, denominator: u32) -> Self {
+        Rational {
+            numerator,
+            denominator,
+        }
+    }
+}
+
+/// Marker trait for differential privacy budgets (regardless of the specific accounting method).
+pub trait DifferentialPrivacyBudget {}
+
+/// Zero-concentrated differential privacy (zCDP) budget.
```

Comment:
Please add a reference to the paper with the standard definition. It can be formatted in markdown, e.g., https://github.com/divviup/libprio-rs/blob/main/src/flp.rs#LL47C1-L47C1